### PR TITLE
enh(API): replace code from repository exception to correct exception class

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description
This PR fix an issue where RepositoryException wasn't used properly. All the methods implemented inside have been moved to a more reliant Exception class. Also the class PlatformTopologyConflictExpcetion has been merged into PlatformTopologyException

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.04.x
- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function apiRequestOnCentralException(string $details): PlatformTopologyRepositoryExceptionInterface
+    {
+        return new self(_("Request to the Central's API failed") . (' : ') . $details);
```
Comment (kduret) [inline src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php:48]:
```suggestion
        return new self(_("Request to the Central's API failed") . ' : ' . $details);
```

File: src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function apiClientException(string $details): PlatformTopologyRepositoryExceptionInterface
+    {
+        return new self(_("API calling the Central returned a Client exception") . (' : ') . $details);
```
Comment (kduret) [inline src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php:56]:
```suggestion
        return new self(_("API calling the Central returned a Client exception") . ' : ' . $details);
```

File: src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function apiRedirectionException(string $details): PlatformTopologyRepositoryExceptionInterface
+    {
+        return new self(_("API calling the Central returned a Redirection exception") . (' : ') . $details);
```
Comment (kduret) [inline src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php:64]:
```suggestion
        return new self(_("API calling the Central returned a Redirection exception") . ' : ' . $details);
```

File: src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function apiServerException(
+        string $message,
+        string $details
+    ): PlatformTopologyRepositoryExceptionInterface {
+        return new self($message . (' : ') . $details);
```
Comment (kduret) [inline src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php:74]:
```suggestion
        return new self($message . ' : ' . $details);
```

File: src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function apiDecodingResponseFailure(string $details): PlatformTopologyRepositoryExceptionInterface
+    {
+        return new self(_("Unable to decode Central's API response") . (' : ') . $details);
```
Comment (kduret) [inline src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php:82]:
```suggestion
        return new self(_("Unable to decode Central's API response") . ' : ' . $details);
```

File: src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function apiUndeterminedError(string $details): PlatformTopologyRepositoryExceptionInterface
+    {
+        return new self(_("Error from Central's register API") . (' : ') . $details);
```
Comment (kduret) [inline src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php:90]:
```suggestion
        return new self(_("Error from Central's register API") . ' : ' . $details);
```

**Review by kduret — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

File: src/Centreon/Domain/PlatformTopology/Interfaces/PlatformTopologyRepositoryExceptionInterface.php [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Centreon\Domain\PlatformTopology\Interfaces;
+
+use Centreon\Domain\Repository\Interfaces\RepositoryExceptionInterface;
+
+interface PlatformTopologyRepositoryExceptionInterface extends RepositoryExceptionInterface
```
Comment (callapa) [inline src/Centreon/Domain/PlatformTopology/Interfaces/PlatformTopologyRepositoryExceptionInterface.php:28]: Just keep this interface empty

File: src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Centreon\Infrastructure\PlatformTopology\Repository\Exception;
+
+use Centreon\Domain\PlatformTopology\Interfaces\PlatformTopologyRepositoryExceptionInterface;
+
+class PlatformTopologyRepositoryException extends \Exception implements PlatformTopologyRepositoryExceptionInterface
```
Comment (callapa) [inline src/Centreon/Infrastructure/PlatformTopology/Repository/Exception/PlatformTopologyRepositoryException.php:28]:
```suggestion
class PlatformTopologyRepositoryException extends RepositoryException implements PlatformTopologyRepositoryExceptionInterface
```

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/Centreon/Application/Controller/PlatformTopologyController.php [hunk 9]
```diff
@@ -132,9 +130,7 @@ public function addPlatformToTopology(Request $request): View
             return $this->view(null, Response::HTTP_CREATED);
         } catch (EntityNotFoundException $ex) {
             return $this->view(['message' => $ex->getMessage()], Response::HTTP_NOT_FOUND);
-        } catch (PlatformTopologyConflictException  $ex) {
-            return $this->view(['message' => $ex->getMessage()], Response::HTTP_CONFLICT);
-        } catch (\Throwable $ex) {
+        } catch (PlatformTopologyException | \Throwable $ex) {
```
[the remark this answers is not in the document]
Comment (jeremyjaouen) [reply src/Centreon/Application/Controller/PlatformTopologyController.php:133]: I have  discuss about this with @kduret . I'm agreed that is not necessary but I found it more readable to keep the details of known Exception that could be catch.

**Review by callapa — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**